File: src/protocol/prompts/chooseTargetAny.ts
import type { TargetingIntent } from "@/types/promptType";

export type Type = "chooseTargetAny";
export type TargetChoice =
  | { kind: "player"; playerId: string }
  | { kind: "card"; cardId: string }
  | { kind: "none" };
export type Input = {
  type: Type;
  validPlayerIds: string[];
  validCardIds: string[];
  hostile?: boolean;
  intent: TargetingIntent;
};
export type Output = { type: "targetAny"; target: TargetChoice };

File: forge-engine/crates/forge-protocol/src/prompts/choose_target_any.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

use crate::prompts::common::TargetAnyChoice;
use crate::values::TargetingIntent;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export, export_to = "prompts/chooseTargetAny.ts")]
pub struct ChooseTargetAnyInput {
    pub valid_player_ids: Vec<String>,
    pub valid_card_ids: Vec<String>,
    #[serde(default)]
    pub hostile: bool,
    pub intent: TargetingIntent,
    pub min_targets: i32,
    pub max_targets: i32,
    pub chosen_targets: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(
    tag = "type",
    rename_all = "camelCase",
    rename_all_fields = "camelCase"
)]
#[ts(export, export_to = "prompts/chooseTargetAny.ts")]
pub enum ChooseTargetAnyOutput {
    TargetAny { target: TargetAnyChoice },
}
